### PR TITLE
[Calling][Bugfix] Skip setup screen logic should be based on redux state

### PR DESCRIPTION
**Pull request by pavelprystinka**
## Purpose

* Skip setup screen logic should be based on redux state

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by ShaunaSong — APPROVED** [review]

**Comment by ahammer** [comment]
Does this need to be 62 files changed,
+916/-1003 lines?

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/handlers/CallStateHandler.kt [hunk 1]
```diff
@@ -45,17 +45,17 @@ internal class CallStateHandler(
...
-        CallingStatus.CONNECTED -> CallCompositeCallStateCode.CONNECTED
-        CallingStatus.CONNECTING -> CallCompositeCallStateCode.CONNECTING
-        CallingStatus.DISCONNECTED -> CallCompositeCallStateCode.DISCONNECTED
-        CallingStatus.DISCONNECTING -> CallCompositeCallStateCode.DISCONNECTING
-        CallingStatus.EARLY_MEDIA -> CallCompositeCallStateCode.EARLY_MEDIA
-        CallingStatus.RINGING -> CallCompositeCallStateCode.RINGING
-        CallingStatus.LOCAL_HOLD -> CallCompositeCallStateCode.LOCAL_HOLD
-        CallingStatus.IN_LOBBY -> CallCompositeCallStateCode.IN_LOBBY
-        CallingStatus.REMOTE_HOLD -> CallCompositeCallStateCode.REMOTE_HOLD
+        CallStatus.CONNECTED -> CallCompositeCallStateCode.CONNECTED
```
Comment (ahammer) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/handlers/CallStateHandler.kt:50]: Please revert the renames in this PR, so it's more manageable for a review.

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt [hunk 2]
```diff
@@ -214,10 +214,11 @@ internal class DependencyInjectionContainerImpl(
...
         AppReduxState(
-            configuration.callConfig?.displayName,
-            localOptions?.isCameraOn == true,
-            localOptions?.isMicrophoneOn == true,
-            localOptions?.audioVideoMode ?: CallCompositeAudioVideoMode.AUDIO_AND_VIDEO
+            displayName = configuration.callConfig?.displayName,
+            cameraOnByDefault = localOptions?.isCameraOn == true,
+            microphoneOnByDefault = localOptions?.isMicrophoneOn == true,
+            avMode = localOptions?.audioVideoMode ?: CallCompositeAudioVideoMode.AUDIO_AND_VIDEO,
+            skipSetupScreen = localOptions?.isSkipSetupScreen ?: false,
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt:221]: Is isSkipSetupScreen a nullable property? Can we check like we did for isCameraOn to keep consistency?

Comment (pavelprystinka) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt:221]: localOptions can be null so `localOptions?` may result in whole `localOptions?.isSkipSetupScreen` be null

**Review by Mohtasim — APPROVED** [review]

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt (hunk 2 above)
Comment (Mohtasim) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt:221]: Sounds good :)

Comment (pavelprystinka) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/di/DependencyInjectionContainerImpl.kt:221]: Discussed with Mohtasim and we decided to `?:` to `isCameraOn` and `isMicrophoneOn` as well

**Review by Mohtasim — APPROVED** [review]

**Review by ahammer — APPROVED** [review]
I'm going to approve, but I'm going to ask to please minimize changes in release branches in the future.

There is still some renames/etc here, and we are supposed to be in "code freeze" at this point except for critical changes. So refactoring/name work should target develop, probably after the merge back.

Please in the future when working in release branch, actively aim to keep the changes as minimal as possible to get the job done.

**Pull request merged**